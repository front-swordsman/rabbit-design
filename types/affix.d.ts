import { ReactNode } from 'react'

export interface AffixProps {
  // 距离窗口顶部达到指定偏移量后触发
  offsetTop?: number
  // 距离窗口底部达到指定偏移量后触发
  offsetBottom?: number
  // 子元素
  children?: ReactNode | String
  // 容器
  target?: () => HTMLElement
  // 固定状态改变时触发的回调函数
  onChange?: (affixed: boolean) => void
  /* 样式相关 */
  className?: string
  // 样式
  style?: React.CSSProperties
}
