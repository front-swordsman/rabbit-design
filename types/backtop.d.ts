export interface BacktopProps {
  /* 滚动高度达到此参数值才出现 BackTop */
  visibilityHeight?: number
  /* 传入子节点 */
  children?: React.ReactNode
  /* 样式相关 */
  className?: string
  /* 样式 */
  style?: React.CSSProperties
}
