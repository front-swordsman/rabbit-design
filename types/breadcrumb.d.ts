export interface BreadcrumbItemProps {
  /** 该级的展示内容 **/
  content?: React.ReactNode
  /** 该级的链接 **/
  link: string
}

export interface BreadcrumbProps {
  /** 组件className前缀 **/
  prefixCls?: string
  /** 样式相关 **/
  className?: string
  /** 样式 **/
  style?: React.CSSProperties
  /** 点击的回调 **/
  onClick?: (index: number, url: string) => void
  /** 尺寸 **/
  size?: 'default' | 'small' | 'large'
  /** 数据源 **/
  dataSource: BreadcrumbItemProps[]
  /** 分隔符 **/
  separator?: React.ReactNode
  /** 最大显示多少级 **/
  expandMax?: number
}
