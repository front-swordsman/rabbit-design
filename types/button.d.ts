
export interface ButtonProps {
  /** 组件className前缀 **/
  prefixCls?: string

  /* 样式相关 */
  // 按钮类名
  className?: string
  // 样式
  style?: React.CSSProperties
  // 按钮尺寸
  size: 'default' | 'small' | 'large'

  /* 功能相关 */
  // 是否禁止按钮
  disabled: false | true
  // 是否正在加载
  loading: false | true
  // 点击的回调
  onClick?: (e: React.MouseEvent) => void
  // 按钮种类
  type: 'primary' | 'gray' | 'warn'
  // 是不是文字按钮
  text: false | true

  /* icon相关 */
  // icon类名
  icon?: string
  // icon样式
  iconStyle?: object

  children?: any
}
