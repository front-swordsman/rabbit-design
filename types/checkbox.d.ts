/**
 * checkbox interface
 */
export interface CheckboxItemProps {
  // 是否被选中 默认为false
  checked?: boolean
  // onChange 变化时候发生的回调
  onChange?: (checked: boolean) => void
  // autofocus 默认聚焦状态
  autoFocus?: boolean
  // disabled 默认为false
  disabled?: boolean
  // className
  className?: string
  // 半选状态 默认为false
  indeterminate?: boolean
  // onBlur
  onBlur?: (e: React.ChangeEvent<HTMLInputElement>) => void
  // onFocus
  onFocus?: (e: React.ChangeEvent<HTMLInputElement>) => void
  // prefixCls
  prefixCls?: string
  children?: React.ReactNode
}

interface OptionValue {
  // 文本信息
  label: React.ReactNode
  // 选中的值
  value: string | number
  // 不可用
  disabled?: boolean
  // autoFocus
  autoFocus?: boolean
}

/**
 * checkbox
 */
export interface CheckboxProps {
  // onChange 默认noop
  onChange?: (checkedValue: Array<string | number>) => void
  // disabled 失效 全部不可用 默认false
  disabled?: boolean
  // 配置的选项 默认为空数组
  options?: OptionValue[]
  // value 指定选中 的项
  value?: Array<string | number>
}