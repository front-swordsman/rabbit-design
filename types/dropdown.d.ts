export type Trigger = 'hover' | 'click'
// dropdown 使用 top, topLeft, topRight, bottomLeft, bottomRight, bottom

// popover and tooltip 使用全部属性
export type Placement = 'top' | 'bottom' | 'topLeft' | 'topRight' | 'bottomLeft' | 'bottomRight'

export interface DropdownProps {
  /** 是否禁用 */
  disabled?: boolean
  // visible 是否可见 默认为 false
  visible?: boolean
  // content dropdown 的内容区
  content?: React.ReactNode
  // wrapperComponent 默认为 span
  wrapperComponent?: any
  // wrapperStyle 包裹层的样式
  wrapperStyle?: React.CSSProperties
  // 触发 dropdown 的方式
  trigger?: Trigger
  // placement 菜单的弹出位置 默认 bottom
  placement?: Placement
  // onVisibleChange 显隐状态变化的回调
  onVisibleChange?: (visible: boolean) => void
  // destroy 是否消失的时候销毁 默认销毁
  destroy?: boolean
  // className 弹出层的 class
  className?: string
  // style dropdown 样式
  style?: React.CSSProperties
  // getPopupContainer
  getPopupContainer?: () => HTMLElement
}
