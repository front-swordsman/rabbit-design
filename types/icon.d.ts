import { HTMLAttributes } from "react";

export interface IconProps {
  // 图标的名称
  type: string
  // size 大小
  size?: number
  // color 图标的颜色
  color?: string
  // rotate 渲染角度
  rotate?: number
  // spin 是否有旋转动画, 默认为false
  spin?: boolean
  // prefixCls 类名的前缀
  prefixCls?: string
  // className
  className?: string
  // style
  style?: React.CSSProperties
  // onClick
  onClick?: (e: React.MouseEvent) => void
}