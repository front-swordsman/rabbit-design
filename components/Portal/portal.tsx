import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { CSSTransition } from 'react-transition-group'
import cx from 'classnames'
import { Portal } from 'types/portal'
import { useClickOutSide, useEnhancedEffect } from '../utils/use'
import { noop } from '../utils/tool'

const defaultProps = {
  trigger: 'hover',
  placement: 'bottomLeft',
  disabled: false,
  wrapperComponent: 'span',
  prefixCls: 'snake-dropdown',
  hasTriangle: false,
  timeout: 300,
  offset: 0,
  mode: 'dropdown',
  visible: false,
  onVisibleChange: noop,
  destroy: true
}

const PortalOverlay: React.FC<Portal> = dropdownProps => {
  const props = { ...defaultProps, ...dropdownProps }
  const {
    className,
    placement,
    content,
    wrapperComponent: Component,
    children,
    trigger,
    disabled,
    prefixCls,
    animationName,
    wrapperStyle,
    style,
    onVisibleChange,
    hasTriangle,
    timeout,
    offset,
    mode,
    visible,
    destroy,
    getPopupContainer
  } = props
  const wrapperRef = React.useRef<HTMLElement>()
  const contentRef = React.useRef<HTMLDivElement>()
  const timeoutRef = React.useRef<any>()
  const [mountNode, setMountNode] = React.useState(null)

  useClickOutSide([contentRef, wrapperRef], () => {
    onVisibleChange(false)
  })

  const toggle = (triggerStr: string, toggle: boolean) => {
    if (disabled) return
    if (trigger === triggerStr) {
      // 只在变化的时候回调
      if (toggle !== visible) {
        onVisibleChange(toggle)
      }
    }
  }

  const getContainer = () => {
    if (getPopupContainer) {
      return getPopupContainer()
    }
    return document.body
  }

  const getOverlayClass = React.useCallback(() => {
    if (animationName) return `${prefixCls}-${animationName}`
    return `${prefixCls}-slide`
  }, [animationName])

  const getContentStyle = (top: string, left: string) => {
    ;(contentRef.current as HTMLElement).style.top = top
    ;(contentRef.current as HTMLElement).style.left = left
  }

  const getPosition = () => {
    const rect = (wrapperRef.current as HTMLElement).getBoundingClientRect()
    const contentRect = (contentRef.current as HTMLElement).getBoundingClientRect()
    let top = `${rect.top + rect.height + offset + window.pageYOffset}px`
    let left = `${rect.left + window.pageXOffset}px`
    if (contentRect.width < rect.width && mode === 'dropdown') {
      contentRef.current.style.minWidth = `${rect.width}px`
    }
    switch (placement) {
      case 'top':
        top = `${rect.top - offset - contentRect.height + window.pageYOffset}px`
        left = `${rect.left + rect.width / 2 - contentRect.width / 2 + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'topLeft':
        top = `${rect.top - offset - contentRect.height + window.pageYOffset}px`
        left = `${rect.left + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'topRight':
        top = `${rect.top - offset - contentRect.height + window.pageYOffset}px`
        left = `${rect.right - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'bottom':
        top = `${rect.top + rect.height + offset + window.pageYOffset}px`
        left = `${rect.left + rect.width / 2 - contentRect.width / 2 + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'bottomLeft':
        top = `${rect.top + rect.height + offset + window.pageYOffset}px`
        left = `${rect.left + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'bottomRight':
        top = `${rect.top + rect.height + offset + window.pageYOffset}px`
        left = `${rect.right - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'left':
        top = `${rect.top + rect.height / 2 - contentRect.height / 2 + window.pageYOffset}px`
        left = `${rect.left - offset - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'leftTop':
        top = `${rect.top + window.pageYOffset}px`
        left = `${rect.left - offset - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'leftBottom':
        top = `${rect.top + rect.height - contentRect.height + window.pageYOffset}px`
        left = `${rect.left - offset - contentRect.width + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'right':
        top = `${rect.top + rect.height / 2 - contentRect.height / 2 + window.pageYOffset}px`
        left = `${rect.right + offset + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'rightTop':
        top = `${rect.top + window.pageYOffset}px`
        left = `${rect.right + offset + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      case 'rightBottom':
        top = `${rect.top + rect.height - contentRect.height + window.pageYOffset}px`
        left = `${rect.right + offset + window.pageXOffset}px`
        getContentStyle(top, left)
        break
      default:
        getContentStyle(top, left)
        break
    }
  }

  const clearTime = () => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current)
    }
  }

  const handleClick = () => {
    toggle('click', true)
  }

  const handleMouseEnter = () => {
    clearTime()
    timeoutRef.current = setTimeout(() => {
      toggle('hover', true)
    }, 0)
  }

  const handleMouseLeave = () => {
    clearTime()
    timeoutRef.current = setTimeout(() => {
      toggle('hover', false)
    }, 0)
  }

  const renderDropDown = () => {
    const classStr = cx(
      prefixCls,
      {
        [`${prefixCls}-${placement}`]: placement,
        [`${prefixCls}-triangle`]: hasTriangle
      },
      className
    )
    return (
      <CSSTransition
        in={visible}
        timeout={timeout}
        classNames={getOverlayClass()}
        unmountOnExit={destroy}
      >
        <div
          className={classStr}
          ref={contentRef}
          onClick={() => {
            onVisibleChange(false)
          }}
          onMouseEnter={handleMouseEnter}
          onMouseLeave={handleMouseLeave}
          style={style}
        >
          <div className={`${prefixCls}-content`}>{content}</div>
        </div>
      </CSSTransition>
    )
  }

  React.useEffect(() => {
    setMountNode(getContainer())
  }, [getContainer()])

  useEnhancedEffect(() => {
    if (visible && mountNode) {
      getPosition()
    }
  }, [visible, mountNode])

  return (
    <>
      <Component
        onClick={handleClick}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        style={wrapperStyle}
        ref={wrapperRef}
      >
        {React.cloneElement(children as any, {
          disabled: disabled
        })}
      </Component>
      {mountNode ? ReactDOM.createPortal(renderDropDown(), getContainer()) : null}
    </>
  )
}

export default PortalOverlay
