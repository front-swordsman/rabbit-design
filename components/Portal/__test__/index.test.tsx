import * as React from 'react'
import { render, fireEvent } from 'react-testing-library'
import Portal from '../index'
import { Placement, Trigger } from 'types/portal'

jest.mock('react-transition-group', () => {
  return {
    CSSTransition: props => {
      return <div>{props.in || !props.unmountOnExit ? props.children : null}</div>
    }
  }
})

describe('Portal Test', () => {
  function PortalDemo({
    placement = 'bottomLeft',
    visible = true,
    disabled = false,
    destroy = true,
    trigger = 'click'
  }: {
    placement?: Placement
    visible?: boolean
    disabled?: boolean
    destroy?: boolean
    trigger?: Trigger
  }) {
    const [show, setShow] = React.useState(visible)
    return (
      <Portal
        visible={show}
        content={<div>内容区</div>}
        placement={placement}
        onVisibleChange={show => setShow(show)}
        trigger={trigger}
        disabled={disabled}
        destroy={destroy}
      >
        <span>click me</span>
      </Portal>
    )
  }
  it('snapshot when visible is false', () => {
    const { baseElement } = render(
      <Portal content={<div>内容区</div>}>
        <span>click me</span>
      </Portal>
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when visible is true', () => {
    const { baseElement } = render(<PortalDemo />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is bottom', () => {
    const { baseElement } = render(<PortalDemo placement="bottom" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is bottomLeft', () => {
    const { baseElement } = render(<PortalDemo placement="bottomLeft" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is bottomRight', () => {
    const { baseElement } = render(<PortalDemo placement="bottomRight" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is top', () => {
    const { baseElement } = render(<PortalDemo placement="top" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is topLeft', () => {
    const { baseElement } = render(<PortalDemo placement="topLeft" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is topRight', () => {
    const { baseElement } = render(<PortalDemo placement="topRight" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is left', () => {
    const { baseElement } = render(<PortalDemo placement="left" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is leftTop', () => {
    const { baseElement } = render(<PortalDemo placement="leftTop" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is leftBottom', () => {
    const { baseElement } = render(<PortalDemo placement="leftBottom" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is right', () => {
    const { baseElement } = render(<PortalDemo placement="right" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is rightTop', () => {
    const { baseElement } = render(<PortalDemo placement="rightTop" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('snapshot when placement is rightBottom', () => {
    const { baseElement } = render(<PortalDemo placement="rightBottom" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('disabled overlay', () => {
    const { getByText, queryByText } = render(<PortalDemo visible={false} disabled />)
    const btn = getByText(/click me/i)
    fireEvent.click(btn)
    expect(queryByText(/内容区/i)).not.toBeInTheDocument()
  })

  it('click show overlay', () => {
    const { getByText, queryByText } = render(<PortalDemo visible={false} />)
    const btn = getByText(/click me/i)
    fireEvent.click(btn)
    expect(queryByText(/内容区/i)).toBeInTheDocument()
    fireEvent.click(queryByText(/内容区/i))
    expect(queryByText(/内容区/i)).not.toBeInTheDocument()
  })

  it('overlay exist with destroy is false', () => {
    const { getByText, queryByText } = render(<PortalDemo visible={false} destroy={false} />)
    const btn = getByText(/click me/i)
    fireEvent.click(btn)
    expect(queryByText(/内容区/i)).toBeInTheDocument()
    fireEvent.click(queryByText(/内容区/i))
    expect(queryByText(/内容区/i)).toBeInTheDocument()
  })
})
