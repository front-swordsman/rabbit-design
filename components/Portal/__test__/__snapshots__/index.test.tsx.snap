// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal Test snapshot when placement is bottom 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-bottom"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is bottomLeft 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-bottomLeft"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is bottomRight 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-bottomRight"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is left 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-left"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is leftBottom 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-leftBottom"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is leftTop 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-leftTop"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is right 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-right"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is rightBottom 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-rightBottom"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is rightTop 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-rightTop"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is top 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-top"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is topLeft 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-topLeft"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when placement is topRight 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-topRight"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Portal Test snapshot when visible is false 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div />
</body>
`;

exports[`Portal Test snapshot when visible is true 1`] = `
<body>
  <div>
    <span>
      <span>
        click me
      </span>
    </span>
  </div>
  <div>
    <div
      class="snake-dropdown snake-dropdown-bottomLeft"
      style="top: 0px; left: 0px;"
    >
      <div
        class="snake-dropdown-content"
      >
        <div>
          内容区
        </div>
      </div>
    </div>
  </div>
</body>
`;
